### PR TITLE
AP-12 check super-parameters conditions

### PR DESCRIPTION
**Pull request by dchertkov**
Added function RowConditions which checks conditions for table row by id and super-parameters conditions

### CODE REVIEW

File: packages/smart/funcs.go [hunk 1]
```diff
@@ -744,6 +752,40 @@ func ColumnCondition(sc *SmartContract, tableName, name, coltype, permissions, i
...
+func RowConditions(sc *SmartContract, tblname string, id int64, validate bool) error {
+	sql := `SELECT conditions FROM ` + converter.EscapeName(getDefTableName(sc, tblname)) + ` WHERE id = ?`
+	condition, err := model.Single(sql, id).String()
+	if err != nil {
+		log.WithFields(log.Fields{"type": consts.DBError, "error": err}).Error("executing single query")
+		return err
+	}
+
+	if len(condition) == 0 {
+		return fmt.Errorf("Item %d has not been found", id)
```
Comment (potehinre) [inline packages/smart/funcs.go]: log with type not found

**Review by potehinre — APPROVED** [review]

File: packages/smart/funcs.go [hunk 2]
```diff
@@ -744,6 +752,41 @@ func ColumnCondition(sc *SmartContract, tableName, name, coltype, permissions, i
 	return sc.AccessTable(tblName, "new_column")
 }
 
+// RowConditions checks conditions for table row by id
+func RowConditions(sc *SmartContract, tblname string, id int64, validate bool) error {
+	sql := `SELECT conditions FROM ` + converter.EscapeName(getDefTableName(sc, tblname)) + ` WHERE id = ?`
+	condition, err := model.Single(sql, id).String()
+	if err != nil {
```
Comment (ghost) [inline packages/smart/funcs.go]: Hide sql queries in package model, please.

**Review by ghost — CHANGES_REQUESTED** [review]

File: packages/smart/funcs.go (hunk 2 above)
Comment (dchertkov) [reply packages/smart/funcs.go]: Fixed in b11cae22f5cf6cf088c3834fbf84df8d91d89348

**Review by ghost — APPROVED** [review]

File: packages/smart/funcs.go [hunk 3]
```diff
@@ -788,6 +796,34 @@ func ColumnCondition(sc *SmartContract, tableName, name, coltype, permissions, i
 	return sc.AccessTable(tblName, "new_column")
 }
 
+// RowConditions checks conditions for table row by id
+func RowConditions(sc *SmartContract, tblname string, id int64) error {
+	esсapedTable := converter.EscapeName(getDefTableName(sc, tblname))
```
Comment (ghost) [inline packages/smart/funcs.go]: escapedTableName

File: packages/smart/funcs.go [hunk 4]
```diff
@@ -788,6 +796,34 @@ func ColumnCondition(sc *SmartContract, tableName, name, coltype, permissions, i
 	return sc.AccessTable(tblName, "new_column")
 }
 
+// RowConditions checks conditions for table row by id
+func RowConditions(sc *SmartContract, tblname string, id int64) error {
```
Comment (ghost) [inline packages/smart/funcs.go:872]: Should be CheckRowConfitions(), I think

**Review by ghost — APPROVED** [review]

File: packages/smart/funcs.go (hunk 4 above)
Comment (dchertkov) [reply packages/smart/funcs.go:872]: May be. But this name was chosen in the style of other TableConditions, ColumnConditions.

**Review by ghost — APPROVED** [review]

**Pull request merged**